<a href="https://colab.research.google.com/github/dcreeder89/Dojo-Project-2/blob/main/Reeder_Project_2_Dataset_Choice.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 Dataset Choice
- Christina Reeder
- 12 Jan 2023

In [8]:
import pandas as pd

## First choice: Heart Failure Prediction

1. Source: [Heart Failure Prediction Dataset](https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction)

In [15]:
url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vRj-4vMvk8k0WT-Ayif1M_gTLOzojgQ7UP_ujcJRSChVDUUrjYNvXkQuHnZplDwKJ0Yqht5p8Ezkbiy/pub?gid=223378998&single=true&output=csv'
df = pd.read_csv(url)

In [16]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [18]:
df.shape

(918, 12)

2. Description of Data

> Dataset to look at risk factors associated with heart disease and whether the patient has been diagnosed with heart disease or not

3. What is the target?

> HeartDisease column

4. What does one row represent?

> A patient

5. Is this a classification or regression problem?

> Classification

6. How many features does this data have?

> 11 features

7. How many rows are in the dataset?

> 918 rows

8. What, if any, challenges do you forsee in cleaning, exploring, or modeling with this dataset?

> I don't forsee any challenges with this dataset. It seems very straight forward and easy to interpret.

## Second choice: Car Insurance

1. Source: [Car Insurance Data](https://www.kaggle.com/datasets/sagnik1511/car-insurance-data)

In [19]:
url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vRZo5fDkeiVEPQ8n2WcK2Xiyo51niu6nfc0uRH5o5ckE_O_9k-5gAxC9xubPFC-IA3s0tr6vAFNJ8Tu/pub?gid=2033371301&single=true&output=csv'
df = pd.read_csv(url)

In [20]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   10000 non-null  int64  
 1   AGE                  10000 non-null  object 
 2   GENDER               10000 non-null  object 
 3   RACE                 10000 non-null  object 
 4   DRIVING_EXPERIENCE   10000 non-null  object 
 5   EDUCATION            10000 non-null  object 
 6   INCOME               10000 non-null  object 
 7   CREDIT_SCORE         9018 non-null   float64
 8   VEHICLE_OWNERSHIP    10000 non-null  int64  
 9   VEHICLE_YEAR         10000 non-null  object 
 10  MARRIED              10000 non-null  int64  
 11  CHILDREN             10000 non-null  int64  
 12  POSTAL_CODE          10000 non-null  int64  
 13  ANNUAL_MILEAGE       9043 non-null   float64
 14  VEHICLE_TYPE         10000 non-null  object 
 15  SPEEDING_VIOLATIONS  10000 non-null  

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0  569520    65+  female  majority               0-9y  high school   
1  750365  16-25    male  majority               0-9y         none   
2  199901  16-25  female  majority               0-9y  high school   
3  478866  16-25    male  majority               0-9y   university   
4  731664  26-39    male  majority             10-19y         none   

          INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0    upper class      0.629027                  1   after 2015        0   
1        poverty      0.357757                  0  before 2015        0   
2  working class      0.493146                  1  before 2015        0   
3  working class      0.206013                  1  before 2015        0   
4  working class      0.388366                  1  before 2015        0   

   CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0         1        10238         12000.0        sedan                    0   
1         0        10238         16000.0        sedan                    0   
2         0        10238         11000.0        sedan                    0   
3         1        32765         11000.0        sedan                    0   
4         0        32765         12000.0        sedan                    2   

   DUIS  PAST_ACCIDENTS  OUTCOME  
0     0               0        0  
1     0               0        1  
2     0               0        0  
3     0               0        0  
4     0               1        1

In [21]:
df.shape

(10000, 19)

2. Description of Data

> Dataset to determine if a car insurance customer will file a claim.

3. What is the target?

> OUTCOME column

4. What does one row represent?

> A customer

5. Is this a classification or regression problem?

> Classification

6. How many features does this data have?

> 18 features

7. How many rows are in the dataset?

> 10,000 rows

8. What, if any, challenges do you forsee in cleaning, exploring, or modeling with this dataset?

> I don't forsee any challenges with this dataset. It seems very straight forward and easy to interpret.